crates/notary/server/src/settings.rs

```diff
@@ -0,0 +1,41 @@
+use config::{Config, ConfigError, Environment, File};
+use std::env;
+use crate::{ CliFields, NotaryServerProperties };
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    #[serde(flatten)]
+    pub config: NotaryServerProperties,
+}
+
+impl Settings {
+    pub fn new(cli_fields: &CliFields) -> Result<Self, ConfigError> {
+        let mut base_path = env::current_dir().expect("Failed to determine the current directory");
+        base_path.pop();
+        let configuration_path = base_path.join("config").join("config.yaml");

```

Comment:
this path should be from `&cli_fields.config_file` as we want to define it there only and not hardcoded here

Reply:
very true, I've updated this to use the `cli_fields`